instance/conn_name.go

```diff
@@ -27,6 +27,8 @@ var (
 	// Additionally, we have to support legacy "domain-scoped" projects
 	// (e.g. "google.com:PROJECT")
 	connNameRegex = regexp.MustCompile("([^:]+(:[^:]+)?):([^:]+):([^:]+)")
+	// The domain name pattern in accordance with RFC 1035, RFC 1123 and RFC 2181.
+	domainNameRegex = regexp.MustCompile("^[A-Za-z0-9-]{1,63}\\.[A-Za-z]{2,6}$")
```

Comment:
This is just placeholder regex as same regex used in [Java Connector](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/595e6206c5e5ff981236662ec466f419c8e97530/core/src/main/java/com/google/cloud/sql/core/CloudSqlInstanceName.java#L54) will not compile in Go.

```
panic: regexp: Compile(`^(?=.{1,255}$)(?!-)[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*\.[A-Za-z]{2,}$`): error parsing regexp: invalid or unsupported Perl syntax: `(?=`
```

Reply:
Using this regex as of now: https://regex101.com/library/SEg6KL